### DISCUSSION TITLE
Bun runtime

### DISCUSSION THREAD

**Discussion by hacknug** [coollabsio/coolify#1215] [Feature Requests]
Now that Bun 1.0 is out the oven, it would be great if we could use it with Coolify. Details for the installation process with Docker can be found here: https://bun.sh/docs/installation

I personally wanna try using it for Nuxt projects so I think this would introduce some complexity in the "Select a Build Pack" view. Using two groups of radios/selects could work better: first you select the base/enviroment and then the specific/preset. Not sure how this is handled in v4 right now (haven't tried in a while and didn't get to far when I did with the first beta).

This was originally proposed by @byawitz in #1054.

**Comment by andrasbacsai** [comment]
I definitely will add it next week. I love Bun, and I would like to use it in other projects.

[Nixpacks](https://github.com/railwayapp/nixpacks/issues/960) also has an issue with adding to its supported language list.

**Comment by andrasbacsai** [comment]
Nixpacks has been updated in beta.44+, so Bun 1.0.0 is available.

**Reply by stefanosandes** [reply]
Hello @andrasbacsai !

I'm trying to deploy a turborepo project that uses bun, but I'm getting this error: bun: command not found.

I just configured "Install command" with `bun install`, otherwise it try to install using npm.

I'm missing something?

Thanks!

**Reply by mrspir** [reply]
> Hello @andrasbacsai !
> 
> I'm trying to deploy a turborepo project that uses bun, but I'm getting this error: bun: command not found.
> 
> I just configured "Install command" with `bun install`, otherwise it try to install using npm.
> 
> I'm missing something?
> 
> Thanks!

Same this error when using bun install command, 

```
ERROR: failed to solve: process "/bin/bash -ol pipefail -c bun install" did not complete successfully: exit code: 127
```

**Reply by AdsonCicilioti** [reply]
On cloud or Self-hosted?

Self-hosted here i got  `command not found` error.

**Reply by mrspir** [reply]
@AdsonCicilioti I'm on Self-hosted

**Reply by AdsonCicilioti** [reply]
One solution I found was to use a Dockerfile instead of `nixpacks`.  Below is an example of a Dockerfile:

```
# Use the official Bun image
FROM oven/bun:1-alpine AS base

WORKDIR /app

# Install dependencies
FROM base AS install
COPY package.json bun.lockb ./
RUN bun install --frozen-lockfile

# Copy source code
FROM base AS prerelease
COPY --from=install /app/node_modules node_modules
COPY . .

# Final stage
FROM base AS release
COPY --from=install /app/node_modules node_modules
COPY --from=prerelease /app .

# Run the app
ENV NODE_ENV=production
USER bun
EXPOSE 3000/tcp
ENTRYPOINT [ "bun", "run", "start" ]

```
[EDIT]

**Reply by elsennov** [reply]
Hi @andrasbacsai , hope you are well. 
I got this error when trying to run Hono App, on Self-Hosted Coolify

```
bun: command not found
```

Is there any solution beside switching to other JS runtime? thank you 🙏

**Reply by hacknug** [reply]
> Hi @andrasbacsai , hope you are well. I got this error when trying to run Hono App, on Self-Hosted Coolify
> 
> ```
> bun: command not found
> ```
> 
> Is there any solution beside switching to other JS runtime? thank you 🙏

Do you have a sandbox or repro repo? I have a project using bun without issues.

Did you add `"packageManager": "bun@1.1.26",` or equivalent to your `package.json`?

**Reply by youknowedo** [reply]
> Do you have a sandbox or repro repo? I have a project using bun without issues.

It depends on the project. Running SvelteKit works fine but if you put it inside a turborepo it fails.

> Is there any solution beside switching to other JS runtime? thank you 🙏

Running `npx bun ...` works fine
